patches/mesonbuild/compilers/detect.py.patch

```diff
@@ -1,8 +1,28 @@
 diff --git a/mesonbuild/compilers/detect.py b/mesonbuild/compilers/detect.py
-index 7542fb628..4dd2dcb22 100644
+index 7bd48d10c..811d8240d 100644
 --- a/mesonbuild/compilers/detect.py
 +++ b/mesonbuild/compilers/detect.py
-@@ -205,6 +205,8 @@ def detect_static_linker(env: 'Environment', compiler: Compiler) -> StaticLinker
+@@ -6,6 +6,7 @@ from __future__ import annotations
+ from ..mesonlib import (
+     MesonException, EnvironmentException, MachineChoice, join_args,
+     search_version, is_windows, Popen_safe, Popen_safe_logged, windows_proof_rm,
++    is_zos,
+ )
+ from ..envconfig import BinaryTable
+ from .. import mlog
+@@ -56,6 +57,11 @@ else:
+         defaults['cpp'] = ['c++', 'g++', 'l++', 'clang++']
+         defaults['objc'] = ['clang']
+         defaults['objcpp'] = ['clang++']
++    elif is_zos():
++        defaults['c'] = ['ibm-clang', 'ibm-clang64']
```

Comment:
If the IBM C/C++ for Open Enterprise Languages on z/OS (https://early-access.ibm.com/software/support/trial/cst/programwebsite.wss?siteId=1803) compiler is installed would it be detected? It provides invocations with the `ibm-` prefix, so `clang`, and `clang64`

Reply:
Added to the default compiler list. The compiler can also be specified manually with the `CC` and `CXX` environment variables or with a [meson native file](https://mesonbuild.com/Machine-files.html)..